MultiplayerMod/Game/UI/Screens/Events/ImmigrantScreenEvents.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using HarmonyLib;
+using MultiplayerMod.Core.Dependency;
+using MultiplayerMod.Core.Patch;
+using MultiplayerMod.Core.Scheduling;
+
+namespace MultiplayerMod.Game.UI.Screens.Events;
+
+[HarmonyPatch(typeof(ImmigrantScreen))]
+[SuppressMessage("ReSharper", "UnusedMember.Local")]
+public static class ImmigrantScreenEvents {
+    public static event Action<List<ITelepadDeliverable>> Initialize;
+    public static event System.Action Reject;
+    public static event Action<ITelepadDeliverable> Proceed;
+
+    [HarmonyPostfix]
+    [HarmonyPatch(nameof(ImmigrantScreen.Initialize))]
+    private static void InitializePatch(ImmigrantScreen __instance) => PatchControl.RunIfEnabled(
+        () => new TaskFactory(Container.Get<UnityTaskScheduler>()).StartNew(
+            async () => {
+                if (ImmigrantScreenPatch.Deliverables != null) return;
+
+                var readyDeliverables = await ImmigrantScreenPatch.WaitForAllDeliverablesReady(__instance);
```

Comment:
Maybe this should be extracted from `ImmigrantScreenPatch`? Just to be consistent with responsibility and refrain patch methods calling from another patch.

Reply:
Done, moved this to ScreenUtils